src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -0,0 +1,369 @@
+
+            var subpatterns = tree.GetRoot().DescendantNodes().OfType<SubpatternSyntax>().ToArray();
+            Assert.Equal(2, subpatterns.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0].NameColon));
+            var x = subpatterns[0].NameColon.Name;
+            var xSymbol = model.GetSymbolInfo(x);
+            Assert.Equal(CandidateReason.None, xSymbol.CandidateReason);
+            Assert.Null(xSymbol.Symbol);
+            Assert.Equal(0, xSymbol.CandidateSymbols.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1].NameColon));
+            var y = subpatterns[1].NameColon.Name;
+            var ySymbol = model.GetSymbolInfo(y);
+            Assert.Equal(CandidateReason.None, ySymbol.CandidateReason);
+            Assert.Null(ySymbol.Symbol);
+            Assert.Equal(0, ySymbol.CandidateSymbols.Length);
+        }
+
+        [Fact]
+        public void PositionalPatternSymbolInfo_01()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        Point p = null;
+        Console.WriteLine(p is ( X: 3, Y: 4 ));
+    }
+}
+class Point
+{
+    public void Deconstruct(out int X, out int Y) => (X, Y) = (3, 4);
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                );
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            var subpatterns = tree.GetRoot().DescendantNodes().OfType<SubpatternSyntax>().ToArray();
+            Assert.Equal(2, subpatterns.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0].NameColon));
+            var x = subpatterns[0].NameColon.Name;
+            var xSymbol = model.GetSymbolInfo(x);
+            Assert.Equal(CandidateReason.None, xSymbol.CandidateReason);
+            Assert.Equal("out System.Int32 X", xSymbol.Symbol.ToTestDisplayString());
+            Assert.Equal(0, xSymbol.CandidateSymbols.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1].NameColon));
+            var y = subpatterns[1].NameColon.Name;
+            var ySymbol = model.GetSymbolInfo(y);
+            Assert.Equal(CandidateReason.None, ySymbol.CandidateReason);
+            Assert.Equal("out System.Int32 Y", ySymbol.Symbol.ToTestDisplayString());
+            Assert.Equal(0, ySymbol.CandidateSymbols.Length);
+        }
+
+        [Fact]
+        public void PositionalPatternSymbolInfo_02()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        Point p = null;
+        Console.WriteLine(p is ( X: 3, Y: 4 ));
```

Comment:
>( X: 3, Y: 4 ) [](start = 31, length = 14)

Nit: the formatting for positional sub-patterns is atypical (here and elsewhere). `p is (X: 3, Y: 4)`